In [86]:
import numpy as np
import pandas as pd

In [87]:
return_df = pd.read_csv('../data/kospi/preprocessed/df_return.csv',index_col=0)*100 #%단위로 환산
abnormal_df = pd.read_csv('../data/return/abnormal_return.csv',index_col=0)
jaccard_df = pd.read_csv('../data/similarity/jaccard_df.csv',index_col=0)
cosine_df = pd.read_csv('../data/similarity/cosine_df.csv',index_col=0)
monthly_df = pd.read_csv('../data/similarity/monthly_sorted_jaccard.csv',index_col=0)

lv2_jaccard = jaccard_df.copy().loc[return_df.index] #거래일에 맞춰서 인덱싱
lv2_cosine = cosine_df.copy().loc[return_df.index]

In [88]:
val_idx,val_col = np.where(np.isfinite(lv2_jaccard))

In [89]:
monthly_df.loc[20110629:,:]

삼양홀딩스      유한양행    CJ대한통운        두산      대림산업  한국테크놀로지그룹  \
20110629  0.482558  0.852459  0.615385  0.579749  0.630503   0.203113   
20110630  0.482558  0.852459  0.615385  0.579749  0.630503   0.203113   
20110701  0.482558  0.852459  0.615385  0.579749  0.630503   0.203113   
20110702  0.482558  0.852459  0.615385  0.579749  0.630503   0.203113   
20110703  0.482558  0.852459  0.615385  0.579749  0.630503   0.203113   
...            ...       ...       ...       ...       ...        ...   
20191227  0.690635  0.532609  0.727488  0.679662  0.510638   0.625402   
20191228  0.690635  0.532609  0.727488  0.679662  0.510638   0.625402   
20191229  0.690635  0.532609  0.727488  0.679662  0.510638   0.625402   
20191230  0.690635  0.532609  0.727488  0.679662  0.510638   0.625402   
20191231  0.690635  0.532609  0.727488  0.679662  0.510638   0.625402   

               기아차  동아쏘시오홀딩스    SK하이닉스        영풍  ...     현대백화점    한국금융지주  \
20110629  0.705706  0.534641  0.750000  0.576200  ...  0.873362  0.663158   
20110630  0.705706  0.534641  0.750000  0.576200  ...  0.873362  0.663158   
20110701  0.705706  0.534641  0.750000  0.576200  ...  0.873362  0.663158   
20110702  0.705706  0.534641  0.750000  0.576200  ...  0.873362  0.663158   
20110703  0.705706  0.534641  0.750000  0.576200  ...  0.873362  0.663158   
...            ...       ...       ...       ...  ...       ...       ...   
20191227  0.678700  0.738854  0.694118  0.811252  ...  0.758364  0.564672   
20191228  0.678700  0.738854  0.694118  0.811252  ...  0.758364  0.564672   
20191229  0.678700  0.738854  0.694118  0.811252  ...  0.758364  0.564672   
20191230  0.678700  0.738854  0.694118  0.811252  ...  0.758364  0.564672   
20191231  0.678700  0.738854  0.694118  0.811252  ...  0.758364  0.564672   

                GS    하나금융지주    아모레퍼시픽        LF        후성   SK이노베이션  \
20110629  0.903475  0.462889  0.577154  0.713235  0.633452  0.574394   
20110630  0.903475  0.462889  0.577154  0.713235  0.633452  0.574394   
20110701  0.903475  0.462889  0.577154  0.713235  0.633452  0.574394   
20110702  0.903475  0.462889  0.577154  0.713235  0.633452  0.574394   
20110703  0.903475  0.462889  0.577154  0.713235  0.633452  0.574394   
...            ...       ...       ...       ...       ...       ...   
20191227  0.671296  0.776720  0.543662  0.870103  0.857909  0.810392   
20191228  0.671296  0.776720  0.543662  0.870103  0.857909  0.810392   
20191229  0.671296  0.776720  0.543662  0.870103  0.857909  0.810392   
20191230  0.671296  0.776720  0.543662  0.870103  0.857909  0.810392   
20191231  0.671296  0.776720  0.543662  0.870103  0.857909  0.810392   

            CJ제일제당      KB금융  
20110629  0.755193  0.716630  
20110630  0.755193  0.716630  
20110701  0.755193  0.716630  
20110702  0.755193  0.716630  
20110703  0.755193  0.716630  
...            ...       ...  
20191227  0.677365  0.821839  
20191228  0.677365  0.821839  
20191229  0.677365  0.821839  
20191230  0.677365  0.821839  
20191231  0.677365  0.821839  

[3108 rows x 113 columns]

In [90]:
lv2_jaccard.iloc[val_idx,:]

삼양홀딩스      유한양행    CJ대한통운  두산      대림산업  한국테크놀로지그룹     기아차  \
20110324       NaN       NaN       NaN NaN       NaN        NaN     NaN   
20110325       NaN       NaN       NaN NaN       NaN        NaN     NaN   
20110328       NaN       NaN       NaN NaN       NaN        NaN     NaN   
20110329       NaN       NaN       NaN NaN       NaN        NaN     NaN   
20110330       NaN  0.852459       NaN NaN       NaN        NaN     NaN   
...            ...       ...       ...  ..       ...        ...     ...   
20190401  0.690635  0.532609  0.727488 NaN  0.510638   0.625402  0.6787   
20190401  0.690635  0.532609  0.727488 NaN  0.510638   0.625402  0.6787   
20190401  0.690635  0.532609  0.727488 NaN  0.510638   0.625402  0.6787   
20190401  0.690635  0.532609  0.727488 NaN  0.510638   0.625402  0.6787   
20190404       NaN       NaN       NaN NaN       NaN        NaN     NaN   

          동아쏘시오홀딩스    SK하이닉스        영풍  ...     현대백화점    한국금융지주        GS  \
20110324       NaN       NaN       NaN  ...       NaN       NaN       NaN   
20110325       NaN       NaN       NaN  ...       NaN       NaN       NaN   
20110328       NaN       NaN       NaN  ...       NaN       NaN       NaN   
20110329       NaN       NaN       NaN  ...       NaN       NaN       NaN   
20110330  0.534641       NaN       NaN  ...       NaN       NaN  0.903475   
...            ...       ...       ...  ...       ...       ...       ...   
20190401  0.738854       NaN  0.811252  ...  0.758364  0.564672  0.671296   
20190401  0.738854       NaN  0.811252  ...  0.758364  0.564672  0.671296   
20190401  0.738854       NaN  0.811252  ...  0.758364  0.564672  0.671296   
20190401  0.738854       NaN  0.811252  ...  0.758364  0.564672  0.671296   
20190404       NaN  0.694118       NaN  ...       NaN       NaN       NaN   

           하나금융지주    아모레퍼시픽        LF        후성   SK이노베이션    CJ제일제당      KB금융  
20110324      NaN       NaN  0.713235       NaN       NaN       NaN       NaN  
20110325      NaN       NaN       NaN       NaN       NaN       NaN       NaN  
20110328      NaN       NaN       NaN       NaN       NaN       NaN       NaN  
20110329      NaN       NaN       NaN       NaN       NaN       NaN       NaN  
20110330      NaN       NaN       NaN       NaN       NaN       NaN       NaN  
...           ...       ...       ...       ...       ...       ...       ...  
20190401  0.77672  0.543662  0.870103  0.857909  0.810392  0.677365  0.821839  
20190401  0.77672  0.543662  0.870103  0.857909  0.810392  0.677365  0.821839  
20190401  0.77672  0.543662  0.870103  0.857909  0.810392  0.677365  0.821839  
20190401  0.77672  0.543662  0.870103  0.857909  0.810392  0.677365  0.821839  
20190404      NaN       NaN       NaN       NaN       NaN       NaN       NaN  

[1017 rows x 113 columns]

In [91]:
monthly_df.iloc[val_idx,:].head(10)

삼양홀딩스  유한양행  CJ대한통운  두산  대림산업  한국테크놀로지그룹  기아차  동아쏘시오홀딩스  SK하이닉스  영풍  \
20101102    NaN   NaN     NaN NaN   NaN        NaN  NaN       NaN     NaN NaN   
20101103    NaN   NaN     NaN NaN   NaN        NaN  NaN       NaN     NaN NaN   
20101104    NaN   NaN     NaN NaN   NaN        NaN  NaN       NaN     NaN NaN   
20101105    NaN   NaN     NaN NaN   NaN        NaN  NaN       NaN     NaN NaN   
20101106    NaN   NaN     NaN NaN   NaN        NaN  NaN       NaN     NaN NaN   
20101106    NaN   NaN     NaN NaN   NaN        NaN  NaN       NaN     NaN NaN   
20101106    NaN   NaN     NaN NaN   NaN        NaN  NaN       NaN     NaN NaN   
20101106    NaN   NaN     NaN NaN   NaN        NaN  NaN       NaN     NaN NaN   
20101106    NaN   NaN     NaN NaN   NaN        NaN  NaN       NaN     NaN NaN   
20101106    NaN   NaN     NaN NaN   NaN        NaN  NaN       NaN     NaN NaN   

          ...  현대백화점  한국금융지주  GS  하나금융지주  아모레퍼시픽  LF  후성  SK이노베이션  CJ제일제당  \
20101102  ...    NaN     NaN NaN     NaN     NaN NaN NaN      NaN     NaN   
20101103  ...    NaN     NaN NaN     NaN     NaN NaN NaN      NaN     NaN   
20101104  ...    NaN     NaN NaN     NaN     NaN NaN NaN      NaN     NaN   
20101105  ...    NaN     NaN NaN     NaN     NaN NaN NaN      NaN     NaN   
20101106  ...    NaN     NaN NaN     NaN     NaN NaN NaN      NaN     NaN   
20101106  ...    NaN     NaN NaN     NaN     NaN NaN NaN      NaN     NaN   
20101106  ...    NaN     NaN NaN     NaN     NaN NaN NaN      NaN     NaN   
20101106  ...    NaN     NaN NaN     NaN     NaN NaN NaN      NaN     NaN   
20101106  ...    NaN     NaN NaN     NaN     NaN NaN NaN      NaN     NaN   
20101106  ...    NaN     NaN NaN     NaN     NaN NaN NaN      NaN     NaN   

          KB금융  
20101102   NaN  
20101103   NaN  
20101104   NaN  
20101105   NaN  
20101106   NaN  
20101106   NaN  
20101106   NaN  
20101106   NaN  
20101106   NaN  
20101106   NaN  

[10 rows x 113 columns]

# Daily

daily announcement effect를 확인할 껀데 공시일 기준 전후 10일간의 CAR을 구할것임. -10일부터 시작하는 건데 이것도 유사도별로 sorting 할 필요가 있음.

논문에서는 Q1, Q5를 표기. 그럼 이걸 어떻게 보여줄 것인가.

예를들어 113개의 기업의 9년동안 유사도가 있음. 2011~2019라고 보자. Q1, Q5의 기업개수는 대충 46개(각23) 정도.

monthly Q1,Q5 index를 구한다 -> 해당 연도에 공시를 한 기업들 확인 -> 그 기업들의 +-10 수익률 저장 -> 평균내기 

In [92]:
monthly_df.loc[20110629:,:]

삼양홀딩스      유한양행    CJ대한통운        두산      대림산업  한국테크놀로지그룹  \
20110629  0.482558  0.852459  0.615385  0.579749  0.630503   0.203113   
20110630  0.482558  0.852459  0.615385  0.579749  0.630503   0.203113   
20110701  0.482558  0.852459  0.615385  0.579749  0.630503   0.203113   
20110702  0.482558  0.852459  0.615385  0.579749  0.630503   0.203113   
20110703  0.482558  0.852459  0.615385  0.579749  0.630503   0.203113   
...            ...       ...       ...       ...       ...        ...   
20191227  0.690635  0.532609  0.727488  0.679662  0.510638   0.625402   
20191228  0.690635  0.532609  0.727488  0.679662  0.510638   0.625402   
20191229  0.690635  0.532609  0.727488  0.679662  0.510638   0.625402   
20191230  0.690635  0.532609  0.727488  0.679662  0.510638   0.625402   
20191231  0.690635  0.532609  0.727488  0.679662  0.510638   0.625402   

               기아차  동아쏘시오홀딩스    SK하이닉스        영풍  ...     현대백화점    한국금융지주  \
20110629  0.705706  0.534641  0.750000  0.576200  ...  0.873362  0.663158   
20110630  0.705706  0.534641  0.750000  0.576200  ...  0.873362  0.663158   
20110701  0.705706  0.534641  0.750000  0.576200  ...  0.873362  0.663158   
20110702  0.705706  0.534641  0.750000  0.576200  ...  0.873362  0.663158   
20110703  0.705706  0.534641  0.750000  0.576200  ...  0.873362  0.663158   
...            ...       ...       ...       ...  ...       ...       ...   
20191227  0.678700  0.738854  0.694118  0.811252  ...  0.758364  0.564672   
20191228  0.678700  0.738854  0.694118  0.811252  ...  0.758364  0.564672   
20191229  0.678700  0.738854  0.694118  0.811252  ...  0.758364  0.564672   
20191230  0.678700  0.738854  0.694118  0.811252  ...  0.758364  0.564672   
20191231  0.678700  0.738854  0.694118  0.811252  ...  0.758364  0.564672   

                GS    하나금융지주    아모레퍼시픽        LF        후성   SK이노베이션  \
20110629  0.903475  0.462889  0.577154  0.713235  0.633452  0.574394   
20110630  0.903475  0.462889  0.577154  0.713235  0.633452  0.574394   
20110701  0.903475  0.462889  0.577154  0.713235  0.633452  0.574394   
20110702  0.903475  0.462889  0.577154  0.713235  0.633452  0.574394   
20110703  0.903475  0.462889  0.577154  0.713235  0.633452  0.574394   
...            ...       ...       ...       ...       ...       ...   
20191227  0.671296  0.776720  0.543662  0.870103  0.857909  0.810392   
20191228  0.671296  0.776720  0.543662  0.870103  0.857909  0.810392   
20191229  0.671296  0.776720  0.543662  0.870103  0.857909  0.810392   
20191230  0.671296  0.776720  0.543662  0.870103  0.857909  0.810392   
20191231  0.671296  0.776720  0.543662  0.870103  0.857909  0.810392   

            CJ제일제당      KB금융  
20110629  0.755193  0.716630  
20110630  0.755193  0.716630  
20110701  0.755193  0.716630  
20110702  0.755193  0.716630  
20110703  0.755193  0.716630  
...            ...       ...  
20191227  0.677365  0.821839  
20191228  0.677365  0.821839  
20191229  0.677365  0.821839  
20191230  0.677365  0.821839  
20191231  0.677365  0.821839  

[3108 rows x 113 columns]

In [93]:
def get_q(df_, lower_bound, upper_bound) : #quntile 설정해주면 해당 quantile에 해당하는 boolean값 반환. cross-sectional 기준
    low_q1_threshold = np.percentile(df_,lower_bound,1)
    low_q1_thres_mat = np.repeat(low_q1_threshold[:,np.newaxis],df_.shape[1],1)

    high_q1_threshold = np.percentile(df_,upper_bound,1)
    high_q1_thres_mat = np.repeat(high_q1_threshold[:,np.newaxis],df_.shape[1],1)
    
    low_bool_mat = low_q1_thres_mat > df_
    high_bool_mat1 = high_q1_thres_mat > df_
    
    low_bool_mat = low_q1_thres_mat > df_
    high_bool_mat = high_q1_thres_mat > df_
    
    return np.bitwise_and(high_bool_mat,~low_bool_mat)

In [94]:
q1_df = get_q(monthly_df.loc[20110629:,:],0,20) #Q1목록
q5_df = get_q(monthly_df.loc[20110629:,:],80,100) #Q5목록

In [95]:
get_after = pd.read_csv('../data/return/get_after.csv',index_col=0)
k_ls = [x for x in get_after.index]
filing_dic = dict(map(lambda x: x.split('_') , k_ls))

dict,key와 value 형태를 만들었으니 이걸로 위 df에서 해당값이 true,false인지 표시하게 하기 -> 그 기업들만 다시 get_after구성 후 누적수익률계산 끝하면 끝.

In [96]:
q1_df.head()

삼양홀딩스   유한양행  CJ대한통운     두산   대림산업  한국테크놀로지그룹    기아차  동아쏘시오홀딩스  \
20110629   True  False   False  False  False       True  False     False   
20110630   True  False   False  False  False       True  False     False   
20110701   True  False   False  False  False       True  False     False   
20110702   True  False   False  False  False       True  False     False   
20110703   True  False   False  False  False       True  False     False   

          SK하이닉스     영풍  ...  현대백화점  한국금융지주     GS  하나금융지주  아모레퍼시픽     LF  \
20110629   False  False  ...  False   False  False    True   False  False   
20110630   False  False  ...  False   False  False    True   False  False   
20110701   False  False  ...  False   False  False    True   False  False   
20110702   False  False  ...  False   False  False    True   False  False   
20110703   False  False  ...  False   False  False    True   False  False   

             후성  SK이노베이션  CJ제일제당   KB금융  
20110629  False    False   False  False  
20110630  False    False   False  False  
20110701  False    False   False  False  
20110702  False    False   False  False  
20110703  False    False   False  False  

[5 rows x 113 columns]

In [99]:
for key,value in filing_dic.items():
    if q1_df.loc[int(value),key] == True:
        print(key,value)

고려제강 20190401
유한양행 20190401
대림산업 20190401
현대건설 20190401
LG상사 20190401
세아베스틸 20190329
롯데푸드 20190329
쌍용양회 20190401
남양유업 20190401
롯데정밀화학 20190401
효성 20190401
SK디스커버리 20190401
삼성SDI 20190329
현대미포조선 20190401
SKC 20190401
HDC 20190401
기업은행 20190401
삼성엔지니어링 20190401
KT 20190401
포스코인터내셔널 20190401
아모레퍼시픽 20190401
부광약품 20190329
